# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/cocoa/nsCursorManager.mm
# Commit: 2424d5c30267
# Full Hash: 2424d5c30267f67ab2b28119d75d89f0441b30e0
# Author: Gijs Kruitbosch <gijskruitbosch@gmail.com>
# Date: 2024-01-23 05:36:48
# Regressor Bug: 1833814
# File Overlap Count: 1
# Description:
#   Bug 1833814 - change nsIWidget's cursor logic, r=emilio,win-reviewers,mhowell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195549
# ==============================================================================

diff -r 94f46aad00df -r 2424d5c30267 widget/cocoa/nsCursorManager.mm
--- a/widget/cocoa/nsCursorManager.mm	Mon Jan 22 20:10:54 2024 +0000
+++ b/widget/cocoa/nsCursorManager.mm	Mon Jan 22 20:42:50 2024 +0000
@@ -276,13 +276,14 @@
 }
 
 - (nsresult)setCustomCursor:(const nsIWidget::Cursor&)aCursor
-          widgetScaleFactor:(CGFloat)scaleFactor {
+          widgetScaleFactor:(CGFloat)scaleFactor
+                forceUpdate:(bool)aForceUpdate {
   NS_OBJC_BEGIN_TRY_BLOCK_RETURN;
 
   // As the user moves the mouse, this gets called repeatedly with the same
   // aCursorImage
-  if (sCurrentCursor == aCursor && sCurrentCursorScaleFactor == scaleFactor &&
-      mCurrentMacCursor) {
+  if (!aForceUpdate && sCurrentCursor == aCursor &&
+      sCurrentCursorScaleFactor == scaleFactor && mCurrentMacCursor) {
     // Native dragging can unset our cursor apparently (see bug 1739352).
     if (MOZ_UNLIKELY(![mCurrentMacCursor isSet])) {
       [mCurrentMacCursor set];